service/drive.py

```diff
@@ -0,0 +1,225 @@
+            .execute()
+        )
+        items = results.get("files", [])
+        return items
+    except HttpError as error:
+        logger.warning(f"An error occurred: {error}")
+        return []
+    
+def make_file_public(service, file_id):
+    """Make a file public."""
+    try:
+        permission = {
+            'type': 'anyone',
+            'role': 'reader',
+        }
+        service.permissions().create(fileId=file_id, body=permission).execute()
+        service.files().get(fileId=file_id, fields='webViewLink').execute()
+        return 'https://www.googleapis.com/drive/v3/files/'+ file_id + '?alt=media&key=' + os.getenv("GCP_API_KEY")
+    except HttpError as error:
+        logger.warning(f"An error occurred while making file Public: {error}")
+        return None
+
+def revoke_public_access_with_retry(service, file_id, max_retries=3):
+    """Attempt to revoke public access, with retries in case of SSL errors."""
+    for attempt in range(max_retries):
+        try:
+            permissions = service.permissions().list(fileId=file_id).execute()
+            for permission in permissions.get('permissions', []):
+                if permission['role'] == 'reader' and permission['type'] == 'anyone':
+                    service.permissions().delete(fileId=file_id, permissionId=permission['id']).execute()
+            break
+        except (HttpError, ssl.SSLEOFError) as e:
+            logger.warning(f"Attempt {attempt + 1} failed for revoke public access: {e}")
+            if attempt < max_retries - 1:
+                time.sleep(2 ** attempt)
+            else:
+                logger.warning("Max retries reached. Could not revoke public access.")
+
+
+def filter_files_by_date(files):
+    """Filter files to include only those created or modified today."""
+    today = datetime.now(timezone.utc).date()
+    filtered_files = []
+    for file in files:
+        created_time = datetime.fromisoformat(file['createdTime'][:-1]).date()
+        modified_time = datetime.fromisoformat(file['modifiedTime'][:-1]).date()
+        if created_time == today:
+            filtered_files.append(file)
+    return filtered_files
+
+ssl_context = ssl.create_default_context()
+ssl_context.set_ciphers("DEFAULT@SECLEVEL=1")
+
+def append_to_sheet(sheets_service, sheet_id, data, retries=3):
+    body = {'values': data}
+    for attempt in range(retries):
+        try:
+            sheets_service.spreadsheets().values().append(
+                spreadsheetId=sheet_id,
+                range="Sheet1!A:C",
+                valueInputOption="RAW",
+                insertDataOption="INSERT_ROWS",
+                body=body
+            ).execute()
+            break
+        except ssl.SSLEOFError:
+            if attempt < retries - 1:
+                time.sleep(2 ** attempt)
+                continue
+            else:
+                raise
+
+def get_transcription_and_summary(file, drive_service, existing_file_ids, new_rows, results):
```

Comment:
What's the reason for returning new_row and using it through reference?

Reply:
For adding new data into sheet,
we need new_row var to collect all the new rows and then merge it into existing sheet with the help of append_to_sheet(sheets_service, sheet_id, new_rows) function